examples/pytorch/correct_and_smooth/model.py

```diff
@@ -0,0 +1,254 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl.function as fn
+
+from dgl.nn.pytorch.conv import GATConv
+
+
+class MLPLinear(nn.Module):
+    def __init__(self, in_dim, out_dim):
+        super(MLPLinear, self).__init__()
+        self.linear = nn.Linear(in_dim, out_dim)
+        self.reset_parameters()
+    
+    def reset_parameters(self):
+        self.linear.reset_parameters()
+
+    def forward(self, x):
+        return F.log_softmax(self.linear(x), dim=-1)
+
+
+class MLP(nn.Module):
+    def __init__(self, in_dim, hid_dim, out_dim, num_layers, dropout=0.):
+        super(MLP, self).__init__()
+        assert num_layers >= 2
+
+        self.linears = nn.ModuleList()
+        self.bns = nn.ModuleList()
+        self.linears.append(nn.Linear(in_dim, hid_dim))
+        self.bns.append(nn.BatchNorm1d(hid_dim))
+
+        for _ in range(num_layers - 2):
+            self.linears.append(nn.Linear(hid_dim, hid_dim))
+            self.bns.append(nn.BatchNorm1d(hid_dim))
+        
+        self.linears.append(nn.Linear(hid_dim, out_dim))
+        self.dropout = dropout
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        for layer in self.linears:
+            layer.reset_parameters()
+        for layer in self.bns:
+            layer.reset_parameters()
+
+    def forward(self, x):
+        for linear, bn in zip(self.linears[:-1], self.bns):
+            x = bn(linear(x))
+            x = F.relu(x, inplace=True)
+            x = F.dropout(x, p=self.dropout, training=self.training)
+        x = self.linears[-1](x)
+        return F.log_softmax(x, dim=-1)
+
+
+class Bias(nn.Module):
+    def __init__(self, size):
+        super().__init__()
+
+        self.bias = nn.Parameter(torch.Tensor(size))
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        nn.init.zeros_(self.bias)
+
+    def forward(self, x):
+        return x + self.bias
+
+
+class GAT(nn.Module):
+    def __init__(self,
+                 in_feats,
+                 n_classes,
+                 n_hidden,
+                 n_layers,
+                 n_heads,
+                 activation,
+                 dropout=0.0,
+                 attn_drop=0.0):
+        super().__init__()
+        self.in_feats = in_feats
+        self.n_classes = n_classes
+        self.n_hidden = n_hidden
+        self.n_layers = n_layers
+        self.num_heads = n_heads
+
+        self.convs = nn.ModuleList()
+        self.linear = nn.ModuleList()
+        self.bns = nn.ModuleList()
+
+        for i in range(n_layers):
+            in_hidden = n_heads * n_hidden if i > 0 else in_feats
+            out_hidden = n_hidden if i < n_layers - 1 else n_classes
+            out_channels = n_heads
+
+            self.convs.append(GATConv(in_hidden,
```

Comment:
Symmetric adjacency normalization is performed in the variant of GATConv: https://github.com/CUAI/CorrectAndSmooth/blob/master/gat/models.py#L152